src/services/location.js

```diff
@@ -47,48 +45,129 @@ ngeo.Location = function(location, history) {
   this.history_ = history;
 
   /**
-   * @type {!goog.Uri}
+   * @type {string|undefined}
    * @private
    */
-  this.uri_ = goog.Uri.parse(location);
-};
+  this.schema_ = location.protocol.substring(0, location.protocol.length - 1);
 
+  /**
+   * @type {string|undefined}
+   * @private
+   */
+  this.domain_ = location.hostname;
 
-/**
- * Get the location's URI object.
- * @return {!goog.Uri} URI.
- * @export
- */
-ngeo.Location.prototype.getUri = function() {
-  return this.uri_;
+  /**
+   * @type {number|undefined}
+   * @private
+   */
+  this.port_ = location.port ? parseInt(location.port, 10) : undefined;
+
+  /**
+   * @type {string|undefined}
+   * @private
+   */
+  this.path_ = location.pathname;
+
+  function fillQueryData(queryData, encodedQuery) {
+    if (encodedQuery) {
+      const pairs = encodedQuery.split('&');
+      for (const pair of pairs) {
+        const indexOfEquals = pair.indexOf('=');
+        if (indexOfEquals >= 0) {
+          const name = pair.substring(0, indexOfEquals);
+          const value = pair.substring(indexOfEquals + 1);
+          queryData[decodeURIComponent(name)] = decodeURIComponent(value);
+        } else {
+          queryData[pair] = '';
+        }
+      }
+    }
+  }
+
+  /**
+   * @type {!Object.<string, string>}
+   * @private
+   */
+  this.queryData_ = {};
+  if (location.search) {
+    fillQueryData(this.queryData_, location.search.substring(1));
+  }
+
+  /**
+   * @type {!Object.<string, string>}
+   * @private
+   */
+  this.fragment_ = {};
+  if (location.hash) {
+    fillQueryData(this.fragment_, location.hash.substring(1));
```

Comment:
Since it is local and not specific to query data. I suggest to rename the function from `fillQueryData` to `fill`.